app/models/fact_value.rb

```diff
@@ -17,17 +17,18 @@ class FactValue < ActiveRecord::Base
   scoped_search :in => :fact_name, :on => :short_name, :complete_value => true, :alias => "fact_short_name"
 
   scope :no_timestamp_facts, lambda {
-    includes(:fact_name).where("fact_names.name <> ?",:_timestamp)
+    includes(:fact_name).where("fact_names.name <> ?",:_timestamp).references(:fact_names)
   }
   scope :timestamp_facts, lambda {
-    joins(:fact_name).where("fact_names.name = ?",:_timestamp)
+    joins(:fact_name).where("fact_names.name = ?",:_timestamp).references(:fact_names)
   }
   scope :my_facts, lambda {
     if !User.current.admin? || Organization.expand(Organization.current).present? || Location.expand(Location.current).present?
```

Comment:
Wouldn't an user get fact values for every host (regardless of taxonomies) this way?